src/backend/utils/adt/agtype_ext.c

```diff
@@ -59,7 +59,7 @@ bool ag_serialize_extended_type(StringInfo buffer, agtentry *agtentry,
         /* copy in the int_value data */
         numlen = sizeof(int64);
         offset = reserve_from_buffer(buffer, numlen);
-        *((int64 *)(buffer->data + offset)) = scalar_val->val.int_value;
+        memcpy( buffer->data + offset, &scalar_val->val.int_value, numlen);
```

Comment:
I don't see why this is necessary, maybe it is just me, could you please elaborate?